src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -190,1352 +330,1149 @@ fd_bpf_loader_v3_user_execute( fd_exec_instr_ctx_t ctx ) {
+      return err;
+    }
+    payer->meta->info.lamports  = payer_lamports;
+    buffer->meta->info.lamports = 0UL;
+
+    /* https://github.com/anza-xyz/agave/blob/574bae8fefc0ed256b55340b9d87b7689bcdf222/programs/bpf_loader/src/lib.rs#L628-L642 */
+    /* Pass an extra account to avoid the overly strict unblanaced instruction error */
+    /* Invoke the system program to create the new account */
     fd_system_program_instruction_create_account_t create_acct;
-    create_acct.lamports = fd_rent_exempt_minimum_balance(ctx.slot_ctx, programdata_len); 
-    create_acct.space = programdata_len;
-    create_acct.owner = ctx.instr->program_id_pubkey;
+    create_acct.lamports = fd_rent_exempt_minimum_balance( instr_ctx->slot_ctx, programdata_len ); 
+    create_acct.space    = programdata_len;
+    create_acct.owner    = instr_ctx->instr->program_id_pubkey;
 
-    fd_system_program_instruction_t instr;
-    instr.discriminant = fd_system_program_instruction_enum_create_account;
+    fd_system_program_instruction_t instr = {0};
+    instr.discriminant         = fd_system_program_instruction_enum_create_account;
     instr.inner.create_account = create_acct;
 
-    /* Setup the accounts passed into the system program create call. Index zero and 
-       one are the from and to accounts respectively for the transfer. The buffer 
-       account is also passed in here. */
     FD_SCRATCH_SCOPE_BEGIN {
-      fd_vm_rust_account_meta_t * acct_metas = (fd_vm_rust_account_meta_t *) 
-                                              fd_scratch_alloc( FD_VM_RUST_ACCOUNT_META_ALIGN, 2 * sizeof(fd_vm_rust_account_meta_t) );
-      fd_native_cpi_create_account_meta( payer_acc, 1, 1, &acct_metas[0] ); 
-      fd_native_cpi_create_account_meta( programdata_acc, 1, 1, &acct_metas[1] );
-      fd_pubkey_t signers[2];
-      ulong signers_cnt = 2;
-      signers[0] = *payer_acc;
-      signers[1] = *programdata_acc;
-
-      err = fd_native_cpi_execute_system_program_instruction( &ctx, &instr, acct_metas, 2, signers, signers_cnt );
-      if (err != 0) {
+      fd_vm_rust_account_meta_t * acct_metas = (fd_vm_rust_account_meta_t*) 
+                                               fd_scratch_alloc( FD_VM_RUST_ACCOUNT_META_ALIGN, 
+                                                                 3UL * sizeof(fd_vm_rust_account_meta_t) );
+      fd_native_cpi_create_account_meta( payer_key,       1U, 1U, &acct_metas[ 0UL ] ); 
+      fd_native_cpi_create_account_meta( programdata_key, 1U, 1U, &acct_metas[ 1UL ] );
+      fd_native_cpi_create_account_meta( buffer_key,      0U, 1U, &acct_metas[ 2UL ] );
+      fd_pubkey_t signers[ 2UL ];
+      ulong signers_cnt = 2UL;
+      signers[ 0UL ] = *payer_key;
+      signers[ 1UL ] = *programdata_key;
+
+      err = fd_native_cpi_execute_system_program_instruction( instr_ctx, &instr, acct_metas, 
+                                                              3UL, signers, signers_cnt );
+      if( err ) {
```

Comment:
```suggestion
      if( FD_UNLIKELY( err ) ) {
```

Do you mind adding this hint to all the conditionals as appropriate?